### PR TITLE
Add Tooljet

### PR DESCRIPTION
**Pull request by Jazys**
Add Tooljet, a tool to build internal dashboard or CRM

### CODE REVIEW

**Comment by Jazys** [comment]
update to pass test on CI/CD

**Comment by Jazys** [comment]
update done

File: apps/tooljet/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+version: "3.7"
+
+services:
+
+  client:
```
Comment (DrMxrcy) [inline apps/tooljet/docker-compose.yml:5]:
```suggestion
  tooljet-client:
```

File: apps/tooljet/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      traefik.http.routers.tooljet-local-insecure.service: tooljet
+      traefik.http.routers.tooljet-local-insecure.middlewares: tooljet-web-redirect
+      # Local domain secure
+      traefik.http.routers.tooljet-local.rule: Host(`tooljet.${LOCAL_DOMAIN}`)
+      traefik.http.routers.tooljet-local.entrypoints: websecure
+      traefik.http.routers.tooljet-local.service: tooljet
+      traefik.http.routers.tooljet-local.tls: true
+    command: openresty -g "daemon off;"
+
+  server:
```
Comment (DrMxrcy) [inline apps/tooljet/docker-compose.yml:44]:
```suggestion
  tooljet-server:
```

File: apps/tooljet/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      - SENTRY_DEBUG    
+      - SSO_DISABLE_SIGNUP
+      - SSO_RESTRICTED_DOMAIN
+      - SSO_GOOGLE_OAUTH2_CLIENT_ID
+      - SSO_GIT_OAUTH2_CLIENT_ID
+      - SSO_GIT_OAUTH2_CLIENT_SECRET
+    command: npm run start:prod
+
+  db-tooljet:
+    container_name: postgres_tooljet
```
Comment (DrMxrcy) [inline apps/tooljet/docker-compose.yml:82]:
```suggestion
    container_name: db-tooljet
```

File: apps/tooljet/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      - tipi_main_network
+    environment:
+      - SERVE_CLIENT=false
+      - TOOLJET_HOST=https://${APP_DOMAIN}
+      - LOCKBOX_MASTER_KEY=70726290922441785304399601987729268981756953900892
+      - SECRET_KEY_BASE=mqrBZqJvzUJ2GsHQaktZjKdb2cvdM8
+      - ORM_LOGGING
+      - PG_DB=tooljet
+      - PG_USER=tooljet
+      - PG_HOST=postgres_tooljet
```
Comment (DrMxrcy) [inline apps/tooljet/docker-compose.yml:60]:
```suggestion
      - PG_HOST=db-tooljet
```

File: apps/tooljet/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  server:
+    container_name: tooljet-server
+    image: tooljet/tooljet-server-ce:v2.14.0    
+    tty: true
+    stdin_open: true
+    restart: always
+    networks:
+      - tipi_main_network
+    environment:
```
Comment (DrMxrcy) [inline apps/tooljet/docker-compose.yml:52]: Remove ENV File, Define Everything Here.

File: apps/tooljet/config.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "../schema.json",
+  "name": "tooljet",
+  "available": true,
+  "exposable": true,
+  "port": 9876,
+  "id": "tooljet",
```
Comment (DrMxrcy) [inline apps/tooljet/config.json:7]:
```suggestion
  "id": "tooljet-client:",
```

Must Match a container Name

File: apps/tooljet/config.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "../schema.json",
+  "name": "tooljet",
```
Comment (DrMxrcy) [inline apps/tooljet/config.json:3]:
```suggestion
  "name": "Tooljet",
```

File: apps/tooljet/app.env [hunk 8]
```diff
@@ -0,0 +1,42 @@
+TOOLJET_HOST=
+
+LOCKBOX_MASTER_KEY=70726290922441785304399601987729268981756953900892
```
Comment (meienberger) [inline apps/tooljet/app.env:3]: These keys should be generated randomly ideally. There is a guide [here](https://www.runtipi.io/docs/contributing/adding-a-new-app#using-a-template-in-your-data-folder) in order to use template files. I can help you with an example if needed

**Review by meienberger — CHANGES_REQUESTED** [review]

**Comment by meienberger** [comment]
Thanks for your PR!

File: apps/tooljet/config.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  "available": true,
+  "exposable": true,
+  "port": 9876,
+  "id": "tooljet",
+  "tipi_version": 23,
+  "version": "2.14.0",
```
Comment (DrMxrcy) [inline apps/tooljet/config.json:9]:
```suggestion
  "version": "v2.4.2",
```

**Review by DrMxrcy — CHANGES_REQUESTED** [review]

**Comment by Jazys** [comment]
Now all fix are made : 
- generate key from random function
- change service and container name

Is it possible to review changes ? @meienberger

**Comment by meienberger** [comment]
Thanks!!

**Pull request merged**